Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,5 +1,61 @@
-function setAlarm() {}
+function formatTimeUnit(unit) {
+  return unit < 10 ? `0${unit}` : unit;
+}
+
+const heading = document.querySelector("#timeRemaining");
+const stopBtn = document.querySelector("#stop");
+heading.innerText = "Time Remaining: 00:00";
+//===========setAlarm======================
+function setAlarm() {
+  const totalTimeInSeconds = parseInt(
+    document.querySelector("#alarmSet").value
+  );
+  setInitialMinutesAndSeconds(heading, totalTimeInSeconds);
+  countDownEverySeconds(totalTimeInSeconds, heading);
+}
+
+//=============setInitialMinutesAndSeconds=======================
+function setInitialMinutesAndSeconds(heading, totalTimeInSeconds) {
+  const initialMinutes = Math.floor(totalTimeInSeconds / 60);
+  const InitialSeconds = totalTimeInSeconds % 60;
+
+  heading.innerText = `Time Remaining: ${formatTimeUnit(
```

Comment:
There are three different places in your code where you basically write ``heading.innerText = `Time Remaining: [some expression to convert to a timestamp]`;``, and there are two places you have the code which takes a number of seconds and turn it into a string that looks like `MM:SS`.

Imagine if you wanted to change part of this. Let's imagine you wanted to change the text "Time Remaining" to "Remaining Time" - you'd need to change three different parts of code. It'd also be easy for you to only change two of them, and miss the third one, and then your UI would change from "Remaining Time" to "Time Remaining" at some point.

Also, imagine if there was a bug in your code which converts seconds to `MM:SS`, if you wanted to show `HH:MM:SS` if the timer was for more than 60 minutes - you would currently need to fix 2-3 different places.

Can you think of a way to only write the code that takes a number of seconds and sets the `heading.innerText` to `MM:SS` one time?

Reply:
This is an excellent observation. I can use a function to handle this.